Made some code cleanup and refactor for the mnist example

dariuszzbyrad opened the pull request:
I think the MNIST example  is somethink like 'Hello World' in learning new programming leangauge, so I made some changes for better readbility and effective learning.

## What changes were proposed in this pull request?
- Code cleanup
- Minor refactor
- Updated use deprecated API
- Add and update some comments

## How was this patch tested?

Running both examples after changes and compare with result from previous version.

RobAltena commented on dl4j-examples/src/main/java/org/deeplearning4j/examples/convolution/mnist/MnistClassifier.java:
```diff
@@ -1,16 +1,10 @@
 package org.deeplearning4j.examples.convolution.mnist;
 
-import java.io.File;
```
In these first lines we can see what you are cleaning up and refactoring. This is good.

RobAltena commented on dl4j-examples/src/main/java/org/deeplearning4j/examples/convolution/mnist/MnistClassifier.java:
```diff
@@ -34,119 +29,137 @@
  */
 public class MnistClassifier {
 
-  private static final Logger log = LoggerFactory.getLogger(MnistClassifier.class);
```
From here we can no longer see what you are cleaning up and refactoring.

dariuszzbyrad replied:
Changed convention name to ALL_CAPS_WITH_UNDERSCORES for static final fields.

dariuszzbyrad commented on dl4j-examples/src/main/java/org/deeplearning4j/examples/convolution/mnist/MnistClassifier.java:
```diff
@@ -34,119 +29,137 @@
-    int outputNum = 10; // 10 digits classification
-    int batchSize = 54;
-    int nEpochs = 1;
-    int iterations = 1;
```
Added descriptions for variables.

dariuszzbyrad commented on dl4j-examples/src/main/java/org/deeplearning4j/examples/convolution/mnist/MnistClassifier.java:
```diff
@@ -34,119 +29,137 @@
-    Random randNumGen = new Random(seed);
-
-    log.info("Data load and vectorization...");
-    String localFilePath = basePath + "/mnist_png.tar.gz";
```
Changed flow. Don't download file if this one exists.

dariuszzbyrad commented on dl4j-examples/src/main/java/org/deeplearning4j/examples/convolution/mnist/MnistClassifier.java:
```diff
@@ -34,119 +29,137 @@
-    DataSetIterator trainIter = new RecordReaderDataSetIterator(trainRR, batchSize, 1, outputNum);
-
-    // pixel values from 0-255 to 0-1 (min-max scaling)
-    DataNormalization scaler = new ImagePreProcessingScaler(0, 1);
```
Changed use constructor with default values min-max (0-1)

dariuszzbyrad commented on dl4j-examples/src/main/java/org/deeplearning4j/examples/convolution/mnist/MnistClassifier.java:
```diff
@@ -34,119 +29,137 @@
-
-    log.info("Network configuration and training...");
-    Map<Integer, Double> lrSchedule = new HashMap<>();
-    lrSchedule.put(0, 0.06); // iteration #, learning rate
```
Added comment with more information about learning rate schedule.

dariuszzbyrad commented on dl4j-examples/src/main/java/org/deeplearning4j/examples/convolution/mnist/MnistClassifier.java:
```diff
@@ -34,119 +29,137 @@
-        .updater(new Nesterovs(new MapSchedule(ScheduleType.ITERATION, lrSchedule)))
-        .weightInit(WeightInit.XAVIER)
-        .list()
-        .layer(0, new ConvolutionLayer.Builder(5, 5)
```
Removed layer index.

dariuszzbyrad commented on dl4j-examples/src/main/java/org/deeplearning4j/examples/convolution/mnist/MnistClassifier.java:
```diff
@@ -34,119 +29,137 @@
-            .nOut(20)
-            .activation(Activation.IDENTITY)
-            .build())
-        .layer(1, new SubsamplingLayer.Builder(SubsamplingLayer.PoolingType.MAX)
```
Removed layer index.

dariuszzbyrad commented on dl4j-examples/src/main/java/org/deeplearning4j/examples/convolution/mnist/MnistClassifier.java:
```diff
@@ -34,119 +29,137 @@
-            .kernelSize(2, 2)
-            .stride(2, 2)
-            .build())
-        .layer(2, new ConvolutionLayer.Builder(5, 5)
```
Removed layer index.

dariuszzbyrad commented on dl4j-examples/src/main/java/org/deeplearning4j/examples/convolution/mnist/MnistClassifier.java:
```diff
@@ -34,119 +29,137 @@
-            .nOut(50)
-            .activation(Activation.IDENTITY)
-            .build())
-        .layer(3, new SubsamplingLayer.Builder(SubsamplingLayer.PoolingType.MAX)
```
Removed layer index.

dariuszzbyrad commented on dl4j-examples/src/main/java/org/deeplearning4j/examples/convolution/mnist/MnistClassifier.java:
```diff
@@ -34,119 +29,137 @@
-            .kernelSize(2, 2)
-            .stride(2, 2)
-            .build())
-        .layer(4, new DenseLayer.Builder().activation(Activation.RELU)
```
Removed layer index.

dariuszzbyrad commented on dl4j-examples/src/main/java/org/deeplearning4j/examples/convolution/mnist/MnistClassifier.java:
```diff
@@ -34,119 +29,137 @@
-            .build())
-        .layer(4, new DenseLayer.Builder().activation(Activation.RELU)
-            .nOut(500).build())
-        .layer(5, new OutputLayer.Builder(LossFunctions.LossFunction.NEGATIVELOGLIKELIHOOD)
```
Removed layer index.

dariuszzbyrad commented on dl4j-examples/src/main/java/org/deeplearning4j/examples/convolution/mnist/MnistClassifier.java:
```diff
@@ -34,119 +29,137 @@
-            .activation(Activation.SOFTMAX)
-            .build())
-        .setInputType(InputType.convolutionalFlat(28, 28, 1)) // InputType.convolutional for normal image
-        .backprop(true).pretrain(false).build();
```
Removed use deprecated methods backprop and pretrain

dariuszzbyrad commented on dl4j-examples/src/main/java/org/deeplearning4j/examples/convolution/mnist/MnistClassifier.java:
```diff
@@ -34,119 +29,137 @@
+        LOGGER.info("The MINIST model has been saved in {}", ministModelPath.getPath());
     }
 
-    ModelSerializer.writeModel(net, new File(basePath + "/minist-model.zip"), true);
```
Added log information with full path to saved model.

dariuszzbyrad commented on dl4j-examples/src/main/java/org/deeplearning4j/examples/convolution/mnist/MnistClassifier.java:
```diff
@@ -34,119 +29,137 @@
-    // vectorization of train data
-    File trainData = new File(basePath + "/mnist_png/training");
-    FileSplit trainSplit = new FileSplit(trainData, NativeImageLoader.ALLOWED_FORMATS, randNumGen);
-    ParentPathLabelGenerator labelMaker = new ParentPathLabelGenerator(); // parent path as the image label
```
Updated comment.

dariuszzbyrad commented on dl4j-examples/src/main/java/org/deeplearning4j/examples/convolution/mnist/MnistClassifier.java:
```diff
@@ -34,119 +29,137 @@
-
-    MultiLayerConfiguration conf = new NeuralNetConfiguration.Builder()
-        .seed(seed)
-        .l2(0.0005)
```
Added comment.

dariuszzbyrad commented on dl4j-examples/src/main/java/org/deeplearning4j/examples/convolution/mnist/MnistClassifierUI.java:
```diff
@@ -1,137 +1,158 @@
-import javafx.scene.shape.StrokeLineCap;
-import javafx.stage.Stage;
-
-/**
```
Updated description.

dariuszzbyrad commented on dl4j-examples/src/main/java/org/deeplearning4j/examples/convolution/mnist/MnistClassifierUI.java:
```diff
@@ -1,137 +1,158 @@
-@SuppressWarnings("restriction")
-public class MnistClassifierUI extends Application {
-
-  private static final String basePath = System.getProperty("java.io.tmpdir") + "/mnist";
```
Changed convention name to ALL_CAPS_WITH_UNDERSCORES for static final fields.

dariuszzbyrad commented on dl4j-examples/src/main/java/org/deeplearning4j/examples/convolution/mnist/MnistClassifierUI.java:
```diff
@@ -1,137 +1,158 @@
-    net = ModelSerializer.restoreMultiLayerNetwork(model);
-  }
-
-  public static void main(String[] args) throws Exception {
```
Removed Exception - doesn't requried.

dariuszzbyrad commented on dl4j-examples/src/main/java/org/deeplearning4j/examples/convolution/mnist/MnistClassifierUI.java:
```diff
@@ -1,137 +1,158 @@
-  }
-
-  @Override
-  public void start(Stage stage) throws Exception {
```
Removed Exception - doesn't requried.

dariuszzbyrad commented on dl4j-examples/src/main/java/org/deeplearning4j/examples/convolution/mnist/MnistClassifierUI.java:
```diff
@@ -1,137 +1,158 @@
-    ImagePreProcessingScaler scaler = new ImagePreProcessingScaler(0, 1);
-    scaler.transform(image);
-    INDArray output = net.output(image);
-    lbl.setText("Prediction: " + net.predict(image)[0] + "\n " + output);
```
Changed showing result for better readability and add log information with full prediction.

dariuszzbyrad commented:
> 
> 
> Please update the PR so that we can review what you have cleaned up and refactored.

Updated. Another changes (not commented) are connected with auto format code.

RobAltena commented:
Oh dear. That is not what I had in mind. I just wanted a clean view under the "files changed" tab and in the git history. I will have a look later.

RobAltena commented:
Thank you for this PR. Appreciated. There is plenty of cleaning up and refactoring to do. We did mess up the git history. But this is the examples repository, not that big a deal here.

dariuszzbyrad commented:
Thanks for the CR. I'll pay more attention to this in the next PR.
